secret-engine/src/main/java/com/sequenceiq/cloudbreak/service/secret/service/SecretService.java

```diff
@@ -140,6 +140,19 @@ public void delete(String secret) {
         LOGGER.trace("Secret delete took {} ms", duration);
     }
 
+    public List<String> listVaultPaths(String secretPathPrefix) {
+        return persistentEngine.listSecretPaths(secretPathPrefix);
+    }
+
+    public void cleanup(String pathPrefix) {
+        metricService.incrementMetricCounter(() -> "secret.cleanup." + pathPrefix);
+        long start = System.currentTimeMillis();
+        engines.stream().forEach(e -> e.cleanup(pathPrefix));
```

Comment:
we should call cleanup on the engine which handles the secret and not on all of them. what happens if we call it on multiple one?

Reply:
I can use "persistentEngine" which handles the secret but the existing code to delete the secret does it on all the engines. I just followed it. If you think it is not needed, I will remove it..